include/swift/Threading/Mutex.h

```diff
@@ -241,6 +241,48 @@ class IndirectMutex {
 using SmallMutex =
     std::conditional_t<sizeof(Mutex) <= sizeof(void *), Mutex, IndirectMutex>;
 
+/// A recursive variant of Mutex.
+class RecursiveMutex {
+
+  RecursiveMutex(const Mutex &) = delete;
+  RecursiveMutex &operator=(const Mutex &) = delete;
+  RecursiveMutex(Mutex &&) = delete;
+  RecursiveMutex &operator=(Mutex &&) = delete;
```

Comment:
I think these want to take `RecursiveMutex` rather than `Mutex` arguments.

Reply:
Copy/pasta trouble, fixed all of these now.